stormpath/express-stormpath
# JSON view model for GET on /register

Fixes #307.

This PR implements a JSON view model for `GET` on `/register` according to the [spec](https://github.com/stormpath/stormpath-framework-spec/blob/master/registration.md#registration-view-model).

This is done by refactoring the `helper/get-login-view-model.js` to be able to take a form for which it generates a view model for (f6c911b).

### Notes

* This PR is going to be merged into `framework-spec-upgrade`.

### How to verify

1. Checkout this branch
2. Clone [express-stormpath-sample-project](https://github.com/stormpath/express-stormpath-sample-project)
3. Use `npm link` and `npm link express-stormpath` to link the library to the example project
4. Make a `GET` request to `/login` with the `Accept` header set to `application/json`
5. Verify that you get a JSON model back according to the [spec](https://github.com/stormpath/stormpath-framework-spec/blob/master/registration.md#registration-view-model)
6. Try to edit the config (`web.register.fields` and `web.register.fieldOrder`) and verify that the fields and its order are correct
7. Verify the content of `accountStores`
8. Try the same for `/login`
